From 8da4d36216eaa94b142155c422a75daafbec0c1f Mon Sep 17 00:00:00 2001
From: Zheng Yejian <zhengyejian1@huawei.com>
Date: Wed, 12 Oct 2022 14:41:42 +0000
Subject: [PATCH 0591/1076] [Huawei] livepatch: Fix warning
 C_RULE_ID_SINGLE_BRANCH_IF_AND_LOOP_BRACKET

Offering: HULK
hulk inclusion
category: bugfix
bugzilla: 187321

--------------------------------

Remove bracket in single branch of if statement.

Signed-off-by: Zheng Yejian <zhengyejian1@huawei.com>
---
 arch/arm/kernel/livepatch.c        | 3 +--
 arch/arm64/kernel/livepatch.c      | 3 +--
 arch/powerpc/kernel/livepatch_32.c | 3 +--
 arch/powerpc/kernel/livepatch_64.c | 3 +--
 4 files changed, 4 insertions(+), 8 deletions(-)

diff --git a/arch/arm/kernel/livepatch.c b/arch/arm/kernel/livepatch.c
index 3379fbf16dd4..a5f4c770990f 100644
--- a/arch/arm/kernel/livepatch.c
+++ b/arch/arm/kernel/livepatch.c
@@ -75,9 +75,8 @@ bool arch_check_jump_insn(unsigned long func_addr)
 	u32 *insn = (u32*)func_addr;
 
 	for (i = 0; i < CHECK_JUMP_RANGE; i++) {
-		if (is_jump_insn(*insn)) {
+		if (is_jump_insn(*insn))
 			return true;
-		}
 		insn++;
 	}
 	return false;
diff --git a/arch/arm64/kernel/livepatch.c b/arch/arm64/kernel/livepatch.c
index 6675569c8a4b..258f1dcda945 100644
--- a/arch/arm64/kernel/livepatch.c
+++ b/arch/arm64/kernel/livepatch.c
@@ -68,9 +68,8 @@ bool arch_check_jump_insn(unsigned long func_addr)
 	u32 *insn = (u32*)func_addr;
 
 	for (i = 0; i < CHECK_JUMP_RANGE; i++) {
-		if (is_jump_insn(*insn)) {
+		if (is_jump_insn(*insn))
 			return true;
-		}
 		insn++;
 	}
 	return false;
diff --git a/arch/powerpc/kernel/livepatch_32.c b/arch/powerpc/kernel/livepatch_32.c
index e45cd3deae08..02c28c40024f 100644
--- a/arch/powerpc/kernel/livepatch_32.c
+++ b/arch/powerpc/kernel/livepatch_32.c
@@ -65,9 +65,8 @@ bool arch_check_jump_insn(unsigned long func_addr)
 	u32 *insn = (u32*)func_addr;
 
 	for (i = 0; i < CHECK_JUMP_RANGE; i++) {
-		if (is_jump_insn(*insn)) {
+		if (is_jump_insn(*insn))
 			return true;
-		}
 		insn++;
 	}
 	return false;
diff --git a/arch/powerpc/kernel/livepatch_64.c b/arch/powerpc/kernel/livepatch_64.c
index 6455be098a23..43f8999aaf4b 100644
--- a/arch/powerpc/kernel/livepatch_64.c
+++ b/arch/powerpc/kernel/livepatch_64.c
@@ -70,9 +70,8 @@ static bool check_jump_insn(unsigned long func_addr)
 	u32 *insn = (u32*)func_addr;
 
 	for (i = 0; i < CHECK_JUMP_RANGE; i++) {
-		if (is_jump_insn(*insn)) {
+		if (is_jump_insn(*insn))
 			return true;
-		}
 		insn++;
 	}
 	return false;
-- 
2.17.1

